Keyboard, sprite and aria support added

icaaq opened the pull request:
Enhanced the code with keyboard support and added aria-states when open/closed.

Added sprite .png and .psd (not sure where you wanted the .psd in the project so I added it to "src/img")

Did not remove the original png

the branch can be looked at here http://icaaq.com/sandbox/mozilla/SocialShare/large.html and here http://icaaq.com/sandbox/mozilla/SocialShare/small.html

My LESS knowledge is less than perfect, so please look if there can be optimizations there.

This will solve [Bug 747963 - Social sharing library: add ability to open/close the window with keyboard]

icaaq committed:
keyboard, sprite and aria support added

Enhanced the code with keyboard support and added aria-states when
open/closed.
Added sprite-image

icaaq committed:
fixed indenting and removed outline

focus and hover state is visible via change of color

icaaq committed:
optimized sprite.png

ran it through ImageOptim

tallowen commented on src/socialshare.js:
```diff
@@ -6,23 +6,19 @@
-    '        <li><a href="https://twitter.com/share" class="twitter-share-button"></a></li>' +
-    '    </ul>' +
-    '</div>' +
+    '    <a class="share-button" href="#" aria-controls="share-options" aria-expanded="false"><span>Share This</span></a>' +
```
I don't think this should be a link - semantically this isn't linking to  another page.

icaaq replied:
Links don't always link to another page, maybe I shall point it to #share-options then it would make more sense. It's a link because I wanted to take advantage of the native keyboard support.

tallowen commented on src/socialshare.js:
```diff
@@ -6,23 +6,19 @@
     var toggle = '[data-toggle="dropdown"]';
     var share_selector = '.socialshare';
-    var fb_iframe = '.socialshare.open[data-type="small-bubbles"] .share-options div.fb-like iframe';
+    var fb_iframe = '.socialshare.open[data-type="small-bubbles"] #share-options div.fb-like iframe';
```
This doesn't allow for multiple plugins on the same page.

icaaq replied:
Hmm, true. I needed a id to point at from the aria-controls attribute. Maybe it's better to change back to class and loop through all .share-options on the page and give them a unique id, like #share-option-x

tallowen commented:
This looks good, thanks for the pull request.

I'm wondering why you added `aria-expanded` attribute the plug-in is already adding the open class to the parent div - is that not enough? it seems to me that It would be better to use one method for determining weather the plug-in is open or closed.

icaaq commented:
<code>aria-expanded</code> tells screen readers that the element is controlling another elements state. Read more about it in the aria spec. 

http://www.w3.org/TR/wai-aria/states_and_properties#aria-expanded

icaaq committed:
spacebar now toggles states

Added ability to toggle open/close states with the spacebar key. Also
added role="button" to the .share-button

icaaq committed:
added aria-hidden and outline

and fixed indentation

Commit:
close widget with escape-key
